contracts/modules/capital/Pool.sol

```diff
@@ -389,16 +420,32 @@ contract Pool is IPool, MasterAwareV2, ReentrancyGuard {
     require(ok, "Pool: Transfer failed");
 
     // Transfer cover assets. Start from 1 (0 is ETH)
-    uint coverAssetsCount = coverAssets.length;
-    for (uint i = 1; i < coverAssetsCount; i++) {
+    uint coverAssetsCount = coverAssets.length - 1;  // substract 1 as ETH is skiped
+    uint investmentAssetsCount = investmentAssets.length;
+    uint totalAssetCount = coverAssetsCount + investmentAssetsCount;
+
+    SwapDetails[] memory _swapDetails = new SwapDetails[](totalAssetCount);
+    Asset[] memory _assets = new Asset[](totalAssetCount);
```

Comment:
feels like passing in the 2 arrays makes things simpler.

On the other side in receiveAssets they materialize in 2 arrays again so this normalizing into 1 array feels like unnecessary complexity since it adds an extra array anyway (isCoverAsset)